### PR TITLE
Documentation, automation, and more

### PR DESCRIPTION
**Pull request by alextremblay**
# Tools

## Nautobot

Improvements to data handling for golden config templates, among others

## Aruba

Improved handling of config data

## Switchconfig

Implemented new config parsing features, built on `uoft_core.BaseSettings`

# Library Improvements

Many bugfixes in `uoft_core`, particularly relating to the new `BaseSettings` class and its `Typer` integration

# Developer Workflow

replaced `monas` with custom invoke tasks and a new script `tasks/init_venv.py`. I'm quite pleased with this one. it can run on any version of python, all the way down to v2.7, and will find or install a python3.10 interpreter and set up a proper virtual environment from it. Will make developer onboarding much much easier

Implemented a tool to make dependency tracking between `uoft_core` and all the projects which use it automatic

Added a ton of additional invoke tasks, as well as documentation in the root `README.md` on how and when to use those tasks

Converted the PR notes from the last PR into actual proper documentation for the uoft_core project, which can be found in `projects/core/README.md`

### CODE REVIEW

File: projects/core/README.md [hunk 1]
```diff
@@ -16,83 +19,138 @@ to include all optional dependencies, run
...
 
 ```
 
-the main api (all the stuff directly importable from `at_utils`) consists of:
-- every function defined in the `main` module
-- the `configure_logging` function from the `log` module
+In this example, I'm debugging the uoft_aruba cli tool. If I run the debug selector, enter "uoft_aruba.cli" in the module prompt, and enter "cpsec whitelist provision some_file.csv" in the args prompt, it would be equivalent to running "uoft_aruba cpsec whitelist provision some_file.csv", but inside of a debugger. This would allow me to debug the `uoft_aruba.cpsec_whitelist.provision` function in the same context as when it's run from the command line in production.
+
+## `uoft_core.debug_cache`
+I've added a handy little utilityu to uoft_core which is great for debugging and testing code which has a long startup (Ex code that loads a bunch of data from an API or SSH session and then processes that data. This `debug_cache`function is a decorator you can add to a function which will save the output of that function to disk the first time that function is run, and then reuse that saved output every subsequent time you run your code.
```
Comment (Kyle-Fozo) [inline projects/core/README.md:137]: utilityu - > utility

Comment (alextremblay) [reply projects/core/README.md:137]: good catch

File: tasks/init_venv.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
```
Comment (Kyle-Fozo) [inline tasks/init_venv.py:1]: I look forward to giving this some test runs

File: projects/core/uoft_core/other.py [hunk 3]
```diff
@@ -269,74 +269,6 @@ def model_to_yaml(model: "BaseModel"):
     return doc
```
Comment (Kyle-Fozo) [inline projects/core/uoft_core/other.py:270]: Will this have any affect on uoft_switchconfig.__main__,py, "from uoft_core.other import Prompt"?

Comment (Kyle-Fozo) [reply projects/core/uoft_core/other.py:270]: Nevermind, excellent :)

**Review by Kyle-Fozo — APPROVED** [review]
LGTM

**Review by Kyle-Fozo — APPROVED** [review]
LGTM

**Review by alextremblay — COMMENTED** [review]
All requested changes complete, PR ready to merge. Appreciate you looking into this

**Pull request merged**